### PR TITLE
Retry on connection failure during joplin ingest

### LINKED ISSUE #170: errors using docker compose commands to start servers
Per the instructions, running

```
docker-compose build
docker-compose up
```

does not seem to work. I can connect to the pgstac server on 8082, but not the sqlalchemy one on 8081, and the sqlalchemy server seems to be constantly restarting. 

I see these errors in the startup log:

```
pgstac-loadjoplin_1     | requests.exceptions.ConnectionError: HTTPConnectionPool(host='app-pgstac', port=8082): Max retries exceeded with url: /collections (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f69d91529a0>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

```
sqlalchemy-migration_1  | requests.exceptions.ConnectionError: HTTPConnectionPool(host='app-sqlalchemy', port=8081): Max retries exceeded with url: /collections (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f09e1e4e9d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

This is using Docker Desktop for MacOSX.

### PR DESCRIPTION
**Pull request by duckontheweb**
Fixes #170

### CODE REVIEW

File: scripts/ingest_joplin.py [hunk 1]
```diff
@@ -5,12 +5,21 @@
...
 joplindata = workingdir.parent / "stac_fastapi" / "testdata" / "joplin"
 
 app_host = sys.argv[1]
 
+retry_strategy = Retry(connect=5, backoff_factor=2)
+adapter = HTTPAdapter(max_retries=retry_strategy)
+
+session = requests.Session()
+session.mount("https://", adapter)
+session.mount("http://", adapter)
```
Comment (kylebarron) [inline scripts/ingest_joplin.py:21]: 🤯 That's so cool!

**Comment by geospatial-jeff** [comment]
I'm guessing the root cause of the issue is the database/application containers becoming available after the ingest script is run?

**Review by geospatial-jeff — APPROVED** [review]

**Comment by duckontheweb** [comment]
> I'm guessing the root cause of the issue is the database/application containers becoming available after the ingest script is run? Adding retry/backoff will fix this.

Yeah, the containers themselves are available, but the FastAPI app within the container must still be booting up in some cases. Maybe the better way to address this is actually using a [`healthcheck`](https://docs.docker.com/compose/compose-file/compose-file-v2/#healthcheck). I'll put in a separate PR with that approach to see how it compares.

**Comment by geospatial-jeff** [comment]
> actually using a healthcheck

I've used this tool before (https://github.com/vishnubob/wait-for-it) in k8s deployments when one container needs to wait for another, could work here.

**Comment by duckontheweb** [comment]
> > actually using a healthcheck
> 
> I've used this tool before (https://github.com/vishnubob/wait-for-it) in k8s deployments when one container needs to wait for another, could work here.

Opened #182 using `wait-for-it.sh` and I like that approach a lot better. Thanks for suggesting that!

**Comment by duckontheweb** [comment]
Closing in favor of #182

**Pull request closed without merging**